lib/ContextResourceString.js

```diff
@@ -60,15 +60,16 @@ ContextResourceString.resClass = "string";
  *
  * @param {String} project the project of the string
  * @param {String} context the context of the string
+ * @param {String} sourcehash the hash valye of the source
  * @param {String} locale the locale of the string
  * @param {String} reskey the key of the string
  * @param {String} datatype the datatype of the string
  * @param {String} flavor the flavor of the app that contains this string
  * @static
  * @return {String} a hash key
  */
-ContextResourceString.hashKey = function(project, context, locale, reskey, datatype, flavor) {
-    var key = ["crs", project, context, locale, reskey, datatype, flavor].join("_");
+ContextResourceString.hashKey = function(project, context, sourcehash, locale, reskey, datatype, flavor) {
```

Comment:
To maintain backward compatibility with existing plugins (and other people's plugins), the new parameter needs to go the end:

```javascript
/**
...
 * @param {String=} sourceHash the optional hash value of the source string
 * @static
 * @return {String} a hash key
 */
ContextResourceString.hashKey = function(project, context, locale, reskey, datatype, flavor, sourceHash) {
    var key = ["crs", project, context, locale, reskey, datatype, flavor, sourceHash].join("_");
    logger.trace("Hashkey is " + key);
    return key;
};
```

Reply:
When you do that, all the rest of the files do not need to change because their API is the same as before. They should operate without the sourceHash parameter like they did before.